### PR TITLE
Test run exit status should be `0` if test retry eventually passed.

### PR DESCRIPTION
**Pull request by xpconanfan**
This change is [Reviewable](https://reviewable.io/reviews/google/mobly/755)

### CODE REVIEW

File: mobly/records.py [hunk 1]
```diff
@@ -607,11 +608,31 @@ def is_test_executed(self, test_name):
         return True
     return False
 
+  def _calculate_retry_count_delta(self):
```
Comment (nkprasad12) [inline mobly/records.py:611]: something along the lines of "eventually_passing_retries" might be a more intuitive name

**Review by nkprasad12 — DISMISSED** [review]

**Review by eric100lin — DISMISSED** [review]
> Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/google/mobly/755)* status: all files reviewed, 1 unresolved discussion (waiting on @nkprasad12 and @xpconanfan)

___
*[mobly/records.py, line 611 at r1](https://reviewable.io/reviews/google/mobly/755#-MbfVngQngbtf64u-Dob-r1-611:-MbkcFT93RENlVIeH05L:b52b0oq) ([raw file](https://github.com/google/mobly/blob/52c5a993e1d592c0147eab466912d9f12914d816/mobly/records.py#L611)):*
<details><summary><i>Previously, nkprasad12 wrote…</i></summary><blockquote>

something along the lines of "eventually_passing_retries" might be a more intuitive name
</blockquote></details>

I agree this comment, maybe `_count_eventually_passing_retries`? 

or do you consider to add another property `is_eventually_pass` instead of changing `is_all_pass` 
and call it at runner side like 
`runner.results.is_all_pass or runner.results.is_eventually_pass and ok`?
(because some tests are actually pass after some retries, 
 not meet "no tests failed or threw errors")

**Review by eric100lin — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/google/mobly/755#-:-MbneTr8DhMTtS9YBTkH:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @nkprasad12 and @xpconanfan)

**Review by xpconanfan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/mobly/755)* status: all files reviewed, 1 unresolved discussion (waiting on @nkprasad12)

___
*[mobly/records.py, line 611 at r1](https://reviewable.io/reviews/google/mobly/755#-MbfVngQngbtf64u-Dob-r1-611:-MbnWfPW5pmz6E-4razO:b-896fix) ([raw file](https://github.com/google/mobly/blob/52c5a993e1d592c0147eab466912d9f12914d816/mobly/records.py#L611)):*
<details><summary><i>Previously, eric100lin (Eric Lin (Tzu Hsiang Lin)) wrote…</i></summary><blockquote>

I agree this comment, maybe `_count_eventually_passing_retries`? 

or do you consider to add another property `is_eventually_pass` instead of changing `is_all_pass` 
and call it at runner side like 
`runner.results.is_all_pass or runner.results.is_eventually_pass and ok`?
(because some tests are actually pass after some retries, 
 not meet "no tests failed or threw errors")
</blockquote></details>

Done.

**Pull request merged**